src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int SIZE_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = defineCell(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> currNode = table[index];
+            while (currNode != null) {
+                if (Objects.equals(key, currNode.key)) {
+                    currNode.value = value;
+                    return;
+                } else if (currNode.next == null) {
+                    currNode.next = newNode;
+                    break;
+                }
+                currNode = currNode.next;
+            }
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[defineCell(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int defineCell(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void resize() {
+        if (size == (int) (DEFAULT_LOAD_FACTOR * table.length)) {
+            transfer(table.length * SIZE_MULTIPLIER);
+        }
+    }
+
+    private void transfer(int newCapacity) {
+        Node<K, V>[] oldTable = table;
+        table = new Node[newCapacity];
+        size = 0;
+        for (Node<K, V> bucket : oldTable) {
+            while (bucket != null) {
+                put(bucket.key, bucket.value);
+                bucket = bucket.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
```

Comment:
what about adding a hash field instead of recalculating it for the existing node?